addons/purchase_stock/models/stock.py

```diff
@@ -216,6 +217,30 @@ def action_view_purchase(self):
 
         return result
 
+    def action_replenish(self):
+        if self.route_id == self.env.ref('purchase_stock.route_warehouse0_buy'):
+            supplier = self.product_id.with_company(self.company_id.id)._select_seller(
+                                        quantity=self.product_min_qty,
+                                        uom_id=self.product_uom)
+            supplier = supplier or self.product_id._prepare_sellers(False).filtered(
```

Comment:
I don't understand this line. 
I select_seller return nothing why do we take a default seller ?